_source/_posts/2021-02-26-unity-webgl-playfab-authorization.md

```diff
@@ -0,0 +1,654 @@
+
+{% img blog/unity-webgl-playfab-authorization/p_soti_stuffing.jpeg alt:"SOTI Credential Stuffing" width:"800" %}{: .center-image }
+
+In their 2020 [State of the Internet / Security](https://www.akamai.com/us/en/multimedia/documents/state-of-the-internet/soti-security-gaming-you-cant-solo-security-report-2020.pdf) report, Akamai tracked 100 Billion credential stuffing attacks from June 2018 to June 2020 and found that 10 Billion of these attacks were targeted at gamers. It is not just up to the player or the distribution platform to take security seriously. Players expect that the game companies producing the product that they are trusting their data to will also be keeping them secure.
+
+In [Identity Security for Games in C# with Unity](/blog/2020/08/21/unity-csharp-games-security), we took a look at the extreme basics of storing and authenticating users in a Unity project. That example is a great starting point for the fundamentals, providing a hello world concept within the Unity editor. I want to expand upon how the design can change depending on the build target and how to make use of a player's authorization by passing it into other back-end platforms. To do this, we will be building a [WebGL](https://docs.unity3d.com/Manual/webgl-gettingstarted.html) application in [Unity](https://unity.com/) that authenticates player's and authorizes them to [Azure Playfab](https://playfab.com/); where their player data will be stored.
+
+{% img blog/unity-webgl-playfab-authorization/p_Unity_Playfab_Auth.png alt:"PlayFab SDK" width:"800" %}{: .center-image }
+
+## Prerequisites
+
+- Sign up for a free [Azure Playfab account](https://developer.playfab.com/en-US/sign-up).
+- Sign up for a free [Azure account](https://azure.microsoft.com/en-us/free/).
+
+- Download and install [Unity](https://unity.com/).
+- Create a new Unity project from Unity Hub with the 3D template
+- Download and configure the [Playfab Unity SDK and Editor Extensions](https://docs.microsoft.com/en-us/gaming/playfab/sdks/unity3d/quickstart)
+
+## Build Target and Design
+
+For this project, I will be targeting [Unity WebGL](https://docs.unity3d.com/Manual/webgl-gettingstarted.html). The build target is important because different platforms will require different means of facilitating user interaction. 
+
+{% img blog/unity-webgl-playfab-authorization/p_state_model.png alt:"Auth Model" width:"800" %}{: .center-image }
```

Comment:
I spent a good 2-3 seconds trying to figure out how this applied to the text above. I suggest moving this image somewhere after you mention the "authentication state machine"

Reply:
Maybe after the text "(account unlock and password reset), etc" you could add something like: "as you can see in the image below, this is quite a lot to implement"

Reply:
Moved it.